packages/cli/snapshots/integration/generators/controller.integration.snapshots.js

```diff
@@ -0,0 +1,371 @@
+      },
+    },
+  })
+  async count(
+    @param.query.object('where', getWhereSchemaFor(ProductReview)) where?: Where<ProductReview>,
+  ): Promise<Count> {
+    return this.barRepository.count(where);
+  }
+
+  @get('/product-reviews', {
+    responses: {
+      '200': {
+        description: 'Array of ProductReview model instances',
+        content: {
+          'application/json': {
+            schema: {type: 'array', items: getModelSchemaRef(ProductReview)},
+          },
+        },
+      },
+    },
+  })
+  async find(
+    @param.query.object('filter', getFilterSchemaFor(ProductReview)) filter?: Filter<ProductReview>,
+  ): Promise<ProductReview[]> {
+    return this.barRepository.find(filter);
+  }
+
+  @patch('/product-reviews', {
+    responses: {
+      '200': {
+        description: 'ProductReview PATCH success count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async updateAll(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(ProductReview, {partial: true}),
+        },
+      },
+    })
+    productReview: ProductReview,
+    @param.query.object('where', getWhereSchemaFor(ProductReview)) where?: Where<ProductReview>,
+  ): Promise<Count> {
+    return this.barRepository.updateAll(productReview, where);
+  }
+
+  @get('/product-reviews/{id}', {
+    responses: {
+      '200': {
+        description: 'ProductReview model instance',
+        content: {'application/json': {schema: getModelSchemaRef(ProductReview)}},
+      },
+    },
+  })
+  async findById(@param.path.number('id') id: number): Promise<ProductReview> {
+    return this.barRepository.findById(id);
+  }
+
+  @patch('/product-reviews/{id}', {
+    responses: {
+      '204': {
+        description: 'ProductReview PATCH success',
+      },
+    },
+  })
+  async updateById(
+    @param.path.number('id') id: number,
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(ProductReview, {partial: true}),
+        },
+      },
+    })
+    productReview: ProductReview,
+  ): Promise<void> {
+    await this.barRepository.updateById(id, productReview);
+  }
+
+  @put('/product-reviews/{id}', {
+    responses: {
+      '204': {
+        description: 'ProductReview PUT success',
+      },
+    },
+  })
+  async replaceById(
+    @param.path.number('id') id: number,
+    @requestBody() productReview: ProductReview,
+  ): Promise<void> {
+    await this.barRepository.replaceById(id, productReview);
+  }
+
+  @del('/product-reviews/{id}', {
+    responses: {
+      '204': {
+        description: 'ProductReview DELETE success',
+      },
+    },
+  })
+  async deleteById(@param.path.number('id') id: number): Promise<void> {
+    await this.barRepository.deleteById(id);
+  }
+}
+
+`;
+
+
+exports[`lb4 controller basic controller scaffolds correct file with args 1`] = `
```

Comment:
nitpick: I feel a bit confused with the names here like what is "args 1" and "input 1", and why they are empty controller class. Maybe add some comments?

Reply:
The keys in the snapshots are generated from the test names, the number at the end is a counter to distinguish between multiple snapshots used by a single test. 

Would it help to use a longer formatting? For example:

```suggestion
exports[`lb4 controller basic controller scaffolds correct file with args -- snapshot 1`] = `
```

Reply:
I am still confused about the `arg 1`, `input 1` here. Could you explain it with an example?

Reply:
Consider the following hypothetical test:

```ts
describe('lb4 datasource', () => {
  it('creates a TS and a JSON file', () => {
    // run a generator that creates two files
    expectFileToMatchSnapshot('src/datasources/db.datasource.ts');
    expectFileToMatchSnapshot('src/datasources/db.datasource.json');
  });
});
```

The are two snapshot-based assertions in a single test, therefore the key used to store & load the snapshot must somehow distinguish between these two.

The solution we (but also Jest and Tap) use: a counter that's increased with each snapshot assertion and that's included in the snapshot key.

In my example code above, the keys will be as follows:
 1. `creates a TS and a JSON file 1` and
 2. `creates a TS and a JSON file 2`.

See the function `buildSnapshotKey`: https://github.com/strongloop/loopback-next/blob/77324c3c78530737893e69df75f4262fb149c161/packages/cli/test/snapshot-matcher.js#L107-L111